charms/istio-gateway/src/charm.py

```diff
@@ -79,12 +84,25 @@ def remove(self, event):
             pilot_port='foo',
         )
 
-        subprocess.run(
-            ["./kubectl", "delete", "-f-"],
-            input=rendered.encode('utf-8'),
-            # Can't remove stuff yet: https://bugs.launchpad.net/juju/+bug/1941655
-            # check=True
-        )
+        try:
+            for obj in codecs.load_all_yaml(rendered):
+                self.lightkube_client.delete(
+                    obj.__class__, obj.metadata.name, namespace=obj.metadata.namespace
```

Comment:
do you like `obj.__class__` or `type(obj)` better here?

Reply:
I've replaced uses of __class__ with type() calls